### ISSUE TITLE
Well... Immedialty some new thoughts.

### ISSUE THREAD

**Issue by Lykdraft** [kmatheussen/radium#1256]
Hey there.:)

**1) Mixer Strips:** 

Maybe it would be a good idea to include a checkbox on a soundobjects right-click menu
to "exclude from showing up in mixer strips".
When opening a mixerstrip even modulators are showing up.
Usually, you have a lot of utility modules that really don't need to clutter up the mixerstrips.

**2) Chase midi events. (Generally known as Midi Chase)**

Really good for long pads, drones, long notes in general. 
What it is (focus on _Midi notes_ only, everything else certainly is cool as well,
but not really _that_ important):

From the Logic Pro X manual :

> If you start playback in the middle of a project, some events might not be heard (such as notes, sustain pedal events, and pitch bend events that start before the point where playback begins). Using the Chase Events function, you can have Logic Pro analyze the project and include some or all of these events when the project plays back.

From the Cubase Manual:

>Chase is a function that makes sure your MIDI instruments sound as they should when you locate to a new position and start playback. This is accomplished by the program transmitting a number of MIDI messages to your instruments each time that you move to a new position in the project, making sure all MIDI devices are set up correctly with regard to program change, controller messages (such as MIDI Volume), etc.

**3) Instrument Panel** 

a) Often there are a LOT of (automatable) parameters (pages after pages) in the instrument panel.
It would be a good idea to have a search filter here.

b) Randomize Parameters: Would be better if Randomize would default to nothing
And the parameters we want to randomize need to be checked. Easier to check 
just a handful of parameters that need to be randomized then unchecking tons of
stuff that shouldn't be.

**4) Sampler:** Spectral Stretch mode in addition to grain?  :)

**Comment by kmatheussen** [comment]
Regarding number 2, effects should already be "chased" in RAdium, as I understand the description. Notes started before the playback point are not started though. I guess that the main issue here?

**Comment by kmatheussen** [comment]
The other issues seems like good ideas. Guess I skip number 4 though as it's a lot of work.

**Comment by kmatheussen** [comment]
(other issues "too" I should have written. Note chasing should be optional though.)

**Comment by kmatheussen** [comment]
"good ideas too".

**Comment by Lykdraft** [comment]
> Notes started before the playback point are not started though. I guess that the main issue here?

Yes, exactly. That was the main point. Yeah, I remember I brought that up with Fx some time ago, as that was even more important.. :) But Note-Chasing is also quite useful as it would allow us to work with very long notes/chords easily.

**Comment by Lykdraft** [comment]
> The other issues seems like good ideas. Guess I skip number 4 though as it's a lot of work.

Yeah, haha. This was just a curveball. I can imagine that this is more on the quite hard--to-do-side of things.

**Comment by kmatheussen** [comment]
It seems more straight forward to implement "note chasing" in logic and programs like that since note end is always defined. In trackers and radium, we often only define when note starts playing, and just let the note stop playing by itself.

**Comment by Lykdraft** [comment]
I don't understand. Note end has nothing to do with this?
It's more or less the same like "poor man's Auto-Seek", just with midi basically.
So it just matters if a note start is somewhere before the actual play-start-point (i.e. note-on is "active").

**Comment by kmatheussen** [comment]
It's much easier to know which notes should not be played if they have a note end.

**Comment by Lykdraft** [comment]
Ah, that way around, I see..

**Comment by kmatheussen** [comment]
At a minimum, it makes sense to not chase-play notes that had note start defined in an earlier played block. That should be a good start.

**Comment by kmatheussen** [comment]
But isn't this implemented already? This is what happens when you enable sample seek for instruments that isn't able to seek (i.e. all instruments except the sample player). Maybe you there should be an option to enable sample seek by default, or even let sample seek be enabled by default?

**Comment by kmatheussen** [comment]
(Sample seek is enabled by checking the "S" button in the instrument widget)

**Comment by Lykdraft** [comment]
Yes, I know, sample seek is implemented. This was just an example how this could work with midi notes.
When I play a pad synth, with notes that are 4 bars long, I always have to start at the beginning of
4 bars to hear the pad. Thats the point. It is cumbersome with long notes. The longer the notes the more time wasted
playing from the start of the notes. Workaround is bouncing/rendering stuff (at least temporarily) to audio, but that's
not ideal. But it's a workaround nonetheless. 

Didn't know that this was such a big deal as this is implemented in nearly all DAWs I know. It's just a convenient thing to have.

**Comment by Lykdraft** [comment]
(When using samples and audio exclusively for this, sample seek would indeed do the trick, but this isn't often the case.)

**Comment by kmatheussen** [comment]
But I'm not talking about sample seek. I'm just saying that enabling sample seek works exactly like "chase midi" when the instrument doesn't support sample seek. And only the sampler instrument supports sample seek. Just try.

**Comment by kmatheussen** [comment]
(you can enable sample seek for all instruments, not just the sampler instrument)

**Comment by Lykdraft** [comment]
> (you can enable sample seek for all instruments, not just the sampler instrument)

Oh really? I'll try that later today for sure.

**Comment by kmatheussen** [comment]
Yeah, you can think of "chase midi" as the fallback mechanism when enabling sample seek and the instrument doesn't support sample seek.

**Comment by Lykdraft** [comment]
Aehm... I am looking for the sample seek setting and somehow I can't find it anymore...  :)
Been a while it seems.... Wasn't there a checkbox somewhere?

**Comment by Lykdraft** [comment]
Nevermind, got it.

**Comment by Lykdraft** [comment]
Yeah it works, great. That's basically all there is to it.
